src/wasm-ir-builder.h

```diff
@@ -211,25 +220,29 @@ class IRBuilder : public UnifiedExpressionVisitor<IRBuilder, Result<>> {
     // The control flow structure we are building expressions for.
     Scope scope;
 
+    // The branch label name for this scope. Always fresh, never shadowed.
+    Name label;
+
     std::vector<Expression*> exprStack;
     // Whether we have seen an unreachable instruction and are in
     // stack-polymorphic unreachable mode.
     bool unreachable = false;
 
     ScopeCtx() : scope(NoScope{}) {}
     ScopeCtx(Scope scope) : scope(scope) {}
+    ScopeCtx(Scope scope, Name label) : scope(scope), label(label) {}
 
     static ScopeCtx makeFunc(Function* func) {
       return ScopeCtx(FuncScope{func});
     }
     static ScopeCtx makeBlock(Block* block) {
       return ScopeCtx(BlockScope{block});
     }
-    static ScopeCtx makeIf(If* iff, Name label = {}) {
-      return ScopeCtx(IfScope{iff, label});
+    static ScopeCtx makeIf(If* iff, Name originalLabel = {}) {
+      return ScopeCtx(IfScope{iff, originalLabel});
     }
-    static ScopeCtx makeElse(If* iff, Name label = {}) {
-      return ScopeCtx(ElseScope{iff, label});
+    static ScopeCtx makeElse(If* iff, Name originalLabel, Name label) {
```

Comment:
what's the relationship between the originalLabel and the label?

Reply:
Yes, exactly 👍